projects/dataset_recorder/data_recorder.cc

```diff
@@ -0,0 +1,246 @@
+// standard includes
+#include<string>
+#include<vector>
+#include <thread>
+#include <mutex>
+#include <atomic>
+
+// BoB includes
+#include "common/main.h"
+#include "common/stopwatch.h"
+#include "common/background_exception_catcher.h"
+#include "common/gps.h"
+#include "common/map_coordinate.h"
+#include "common/bn055_imu.h"
+
+#include "imgproc/opencv_unwrap_360.h"
+#include "video/panoramic.h"
+
+#include <sys/stat.h>
+#include <ctime>
+
+
+
+
+
+
+// Standard C++ includes
+#include <iostream>
+
+using namespace BoBRobotics;
+using namespace BoBRobotics::ImgProc;
+using namespace BoBRobotics::Video;
+
+
+#define WIDTH 192
+#define HEIGHT 72
+
+#define IS_UNWRAP true
+#define UPDATE_INTERVAL 200
+
+
+void readCameraThreadFunc(BoBRobotics::Video::Input *cam,
+                        cv::Mat &img,
+                        std::atomic<bool> &shouldQuit,
+                        std::mutex &mutex)
+{
+
+    //auto unwrapper = cam->createUnwrapper(cv::Size(WIDTH,HEIGHT));
+
+    while(!shouldQuit) {
+        std::this_thread::sleep_for(std::chrono::milliseconds(5));
+        std::lock_guard<std::mutex> lock(mutex);
+        cam->readFrame(img);
+
+    }
+
+
+}
+
+void saveThread(cv::Mat img, std::string fileName) {
+    cv::imwrite(fileName, img);
+}
+
+int bobMain(int argc, char* argv[])
+{
+
+    using degree_t = units::angle::degree_t;
+    std::vector<cv::Mat> images;
+    std::vector<std::string> fileNames;
+
+    // setting up gps
+    const char *path_linux = "/dev/ttyACM1"; // path for linux systems
+    BoBRobotics::GPS::Gps gps;
+    gps.connect(path_linux);
+    BoBRobotics::BN055 imu;
+    const int maxTrials = 3;
+    int numTrials = maxTrials;
+    std::atomic<bool> shouldQuit{false};
+    std::mutex mex;
+
+
+    time_t rawtime;
+    struct tm * timeinfo;
+    char buffer[80];
+
+    time (&rawtime);
+    timeinfo = localtime(&rawtime);
+
+    strftime(buffer,sizeof(buffer),"%d-%m-%Y %H:%M:%S",timeinfo);
+    std::string str(buffer);
```

Comment:
We've talked about the Jetson's clock issue, but let's leave this bit as is for now and raise it as a Github issue after it's merged.